cli/src/stake.rs

```diff
@@ -1746,10 +1746,11 @@ pub fn process_show_stake_history(
     config: &CliConfig,
     use_lamports_unit: bool,
 ) -> ProcessResult {
-    let stake_history_account = rpc_client.get_account(&stake_history::id())?;
-    let stake_history = from_account::<StakeHistory>(&stake_history_account).ok_or_else(|| {
-        CliError::RpcRequestError("Failed to deserialize stake history".to_string())
-    })?;
+    let stake_history_account = rpc_client.get_account_shared_data(&stake_history::id())?;
```

Comment:
Are we going to expose the new structures to rpc_client's or keep them on `Account`?

Reply:
Perhaps I need to take another path at detangling things. It seemed there was cross use where some number of callers in our mono repo were calling get_account. It seems that anything like the cli which perhaps could make copies could benefit from the arc ideas or the differing storage formats internally? I guess our cli is considered an rpc client? And would you rather keep those as Account?

Reply:
Basically, anything that ends up calling any shared code should ideally be using AccountSharedData, or else we could end up with conversions, which could become hidden copies at every intersection of the types.

Reply:
+1 on reducing copies, @CriesofCarrots what are your thoughts on rpc_client switching to `AccountSharedData`?

Reply:
Our cli uses rpc_client under the hood, yes.
I'm not sure I see a strong need to use `AccountSharedData` in rpc_client. The account has to be copied to be transmitted via rpc regardless, with allocation on the receiving end. Essentially the main benefit the duplicated methods provide is performing a conversion to `AccountSharedData`, and I wonder if that, or other Arc wrapping, should be explicit on the part of the caller? (Fwiw, I haven't thought of a viable use-case for WritableAccounts in clients.)

That said, the only real harm in supporting the additional methods is the added complexity of choosing when to use one or the other. So if we go that route, I'd like to see thorough documentation on how a client should make that choice.

Reply:
Agree, no reason to extend this type through the rpc boundary at this moment.  Jeff, how much work is it to pull this back a bit and keep AccountSharedData on the "inside"?